client/app/hosting/framework/runtime/hosting-framework-runtime.controller.js

```diff
@@ -0,0 +1,112 @@
+angular.module("App").controller(
+    "HostingFrameworkRuntimeCtrl",
+    class HostingFrameworkRuntimeCtrl {
+
+        /**
+         * @constructs HostingFrameworkRuntimeCtrl
+         * @param $scope
+         * @param $stateParams
+         * @param $timeout
+         * @param Alerter
+         * @param Hosting
+         * @param HostingFrameworkRuntime
+         */
+        constructor ($scope, $stateParams, $timeout, Alerter, Hosting, HostingFrameworkRuntime) {
+            this.$scope = $scope;
+            this.$stateParams = $stateParams;
+            this.$timeout = $timeout;
+            this.Alerter = Alerter;
+            this.Hosting = Hosting;
+            this.HostingFrameworkRuntime = HostingFrameworkRuntime;
+        }
+
+        /**
+         * Initialize HostingFrameworkRuntimeCtrl
+         */
+        $onInit () {
+            this.hasResult = false;
+            this.loading = true;
+            this.runtimes = [];
+            this.maxRuntimes = 0;
+
+            this.$scope.$on(this.Hosting.events.tabFrameworkRuntimesRefresh, () => this.getIds());
+
+            this.getIds();
+            this.loadCapabilities();
+        }
+
+        /**
+         * Load all runtimes ids from API
+         */
+        getIds () {
+            return this.HostingFrameworkRuntime.list(this.$stateParams.productId)
+                .then((ids) => {
+                    if (ids) {
+                        const runtimeIds = ids.sort((a, b) => a < b);
+
+                        this.runtimes = runtimeIds.map((id) => ({ id }));
+                    }
+                })
+                .catch((err) => {
+                    this.Alerter.error(this.$scope.tr("hosting_tab_FRAMEWORK_runtime_list_error") + err.message, this.$scope.alerts.main);
+                })
+                .finally(() => {
+                    if (this.runtimes.length > 0) {
+                        this.hasResult = true;
```

Comment:
`this.hasResult = _.isArray(this.runtimes) && !_.isEmpty(this.runtimes)`

Reply:
or this.runtimes.length, but I would completely avoid using hasResult